src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -680,6 +680,12 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 "An empty indexer is not allowed. Specify a valid expression."
             );
 
+            public Diagnostic RuntimePropertyNotAllowed(string property) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP118",
+                $"The property \"{property}\" cannot be assigned a runtime value. Accessible resource properties are \"id\", \"name\", and \"type\" and module properties are \"name\""

```

Comment:
This feels like something which could get out of sync with the code. Please either find a way to avoid this happening (e.g. common place in the code which won't get missed on a refactor), or avoid specifying the properties in the message. The former is obviously preferred if possible.

Reply:
This could be more messy than necessary but possible. We'd have to figure out what `DeclaredSymbol` we're trying to access, collect all its `DeployTimeConstant` field names and then pass those to the Exception. Let me know if that's worth doing.

Reply:
This is now implemented. Properties will be programmatically be generated into the error message